doc/source/flexibility/binlogging_replication_improvements.rst

```diff
@@ -118,6 +118,158 @@ about what else affects the writing of ``FLUSH`` commands to the binary log.
+      SELECT get_last_gtid_from_binlog("binlog.00001") AS result;
+      +-----------------------------------------+
+      | result                                  |
+      +=========================================+
+      | F6F54186-8495-47B3-8D9F-011DDB1B65B3:10 |
+      +-----------------------------------------+
+
+      DROP FUNCTION get_last_gtid_from_binlog;
+
+The basic syntax for ``get_gtid_set_by_binlog()`` is the following:
+
+* get_gtid_set_by_binlog(string) [AS] alias
+
+  Usage: SELECT get_gtid_set_by_binlog(string) [AS] alias
+
+  Example:
+
+  .. code-block:: mysql
+
+      CREATE FUNCTION get_gtid_set_by_binlog RETURNS STRING SONAME 'libbinlog_utils_udf.so';
+      SELECT get_gtid_set_by_binlog("binlog.00001") AS result;
+      +-------------------------+
+      | result                  |
+      +=========================+
+      | 11ea-b9a7:7,11ea-b9a7:8 |
+      +-------------------------+
+
+      DROP FUNCTION get_gtid_set_by_binlog;
+
+The basic syntax for ``get_binlog_by_gtid_set()`` is the following:
+
+* get_binlog_by_gtid_set(string) [AS] alias
+
+  Usage: SELECT get_binlog_by_gtid_set(string) [AS] alias
+
+  Example:
+
+  .. code-block:: mysql
+
+      CREATE FUNCTION get_binlog_by_gtid_set RETURNS STRING SONAME 'libbinlog_utils_udf.so';
+      SELECT get_binlog_by_gtid_set("11ea-b9a7:7,11ea-b9a7:8") AS result;
+      +---------------------------------------------------------------+
+      | result                                                        |
+      +===============================================================+
+      | bin.000003/11ea-b9a7:9,11ea-b9a7:10,11ea-b9a7:11,11ea-b9a7:12 |
+      +---------------------------------------------------------------+
+
+      DROP FUNCTION get_binlog_by_gtid_set;
+
+The basic syntax for ``get_first_record_timestamp_by_binlog()`` is the following:
+
+* get_first_record_timestamp_by_binlog(TIMESTAMP) [AS] alias
+
+  Usage: SELECT get_first_record_timestamp_by_binlog(TIMESTAMP) [AS] alias
+
+  Example:
+
+  .. code-block:: mysql
+
+      CREATE FUNCTION SELECT get_first_record_timestamp_by_binlog RETURNS TIMESTAMP SONAME 'libbinlog_utils_udf.so';
+      SELECT SELECT get_first_record_timestamp_by_binlog("bin.00003") AS result;
+      +---------------------+
+      | result              |
+      +=====================+
+      | 2020-12-03 09:10:40 |
+      +---------------------+
+
+      DROP FUNCTION get_first_record_timestamp_by_binlog;
+
+The basic syntax for ``get_last_record_timestamp_by_binlog()`` is the following:
+
+* get_last_record_timestamp_by_binlog(TIMESTAMP) [AS] alias
+
+  Usage: SELECT get_last_record_timestamp_by_binlog(TIMESTAMP) [AS] alias
+
+  Example:
+
+  .. code-block:: mysql
+
+      CREATE FUNCTION SELECT get_last_record_timestamp_by_binlog RETURNS TIMESTAMP SONAME 'libbinlog_utils_udf.so';
+      SELECT SELECT get_last_record_timestamp_by_binlog("bin.00003") AS result;
```

Comment:
Likewise, similar to `get_first_record_timestamp_by_binlog()`

Reply:
Updated - removed second SELECT